### PR TITLE
Remove Datastore's Serializable class and related changes

### LINKED ISSUE #1112: remove newlines from datastore Key.toString() values
Calling `.toString()` on a datastore `Key`, for instance when logging some value or operation in application code, returns a String containing newlines and tabs, which might be too verbose in places where each "log line" is expected to be a single line.

For instance:

```
partition_id {
  project_id: "some-project-id"
}
path {
  kind: "taskqueue"
  id: -1037827333
}
```

Since the toString implemented in BaseKey is converting the instance to protobuf and calling toString on the returned GeneratedMessage, could one of the protobuf methods for formatting without newlines be used instead for condensed output?

### PR DESCRIPTION
**Pull request by mziccard**
This fixes #1112

### CODE REVIEW

**Comment by mziccard** [comment]
This removes Datastore's `Serializable` class (as I mentioned this some time ago). Other than that this PR also:
- Implements `toString` on a per-class basis without relying on `toPb`
- Removes `to/fromPb` methods that are not needed
- Fixes some minor style issues

@aozarov your opinion would be valuable in this

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/BaseEntity.java [hunk 1]
```diff
@@ -49,12 +49,10 @@
  * @see <a href="https://cloud.google.com/datastore/docs/concepts/entities">Google Cloud Datastore
  *     Entities, Properties, and Keys</a>
  */
-public abstract class BaseEntity<K extends IncompleteKey>
-    extends Serializable<com.google.datastore.v1.Entity> {
+public abstract class BaseEntity<K extends IncompleteKey> implements java.io.Serializable {
```
Comment (aozarov) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/BaseEntity.java]: import `java.io.Serializable`?

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/Cursor.java [hunk 2]
```diff
@@ -22,17 +22,18 @@
 import com.google.common.base.MoreObjects.ToStringHelper;
 import com.google.common.io.BaseEncoding;
 import com.google.protobuf.ByteString;
-import com.google.protobuf.InvalidProtocolBufferException;
+
```
Comment (aozarov) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/Cursor.java]: FYI, The google style guide has changed/relaxed regarding [imports](https://google.github.io/styleguide/javaguide.html#s3.3.3-import-ordering-and-spacing).

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/Cursor.java [hunk 3]
```diff
@@ -22,17 +22,18 @@
...
+import java.io.Serializable;
 
 /**
  * A Google Cloud Datastore cursor.
  * The cursor can be used to as a starting point or an ending point for a {@link Query}
  */
-public final class Cursor extends Serializable<com.google.datastore.v1.Value> {
+public final class Cursor implements Serializable {
```
Comment (aozarov) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/Cursor.java]: A toString would be nice here.

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/DateTime.java [hunk 4]
```diff
@@ -32,12 +31,11 @@
  * @see <a href="https://cloud.google.com/datastore/docs/concepts/entities">Google Cloud Datastore
  *     Entities, Properties, and Keys</a>
  */
-public final class DateTime extends Serializable<com.google.protobuf.Timestamp>
-    implements Comparable<DateTime> {
+public final class DateTime implements Comparable<DateTime>, Serializable {
```
Comment (aozarov) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/DateTime.java]: A toString would be nice here.

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/Cursor.java (hunk 3 above)
Comment (mziccard) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/Cursor.java]: There already is one, see line 53.

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/DateTime.java (hunk 4 above)
Comment (mziccard) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/DateTime.java]: Also here, one was already there, see line 45.

Comment (aozarov) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/DateTime.java]: Oh, right. fooled by partial collapse.

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/StructuredQuery.java [hunk 5]
```diff
@@ -752,9 +751,24 @@ B mergeFrom(com.google.datastore.v1.Query queryPb) {
   }
 
   @Override
+  public String toString() {
+    return super.toStringHelper()
```
Comment (aozarov) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/StructuredQuery.java]: Nit, unnecessary call to `super`.

**Comment by aozarov** [comment]
LGTM

**Pull request merged**